src/v/coproc/api/wasm_list.cc

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright 2021 Vectorized, Inc.
+ *
+ * Licensed as a Redpanda Enterprise file under the Redpanda Community
+ * License (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * https://github.com/vectorizedio/redpanda/blob/master/licenses/rcl.md
+ */
+
+#include "coproc/api/wasm_list.h"
+
+#include "model/record_batch_types.h"
+#include "rapidjson/document.h"
+#include "reflection/adl.h"
+#include "storage/record_batch_builder.h"
+
+#include <seastar/core/coroutine.hh>
+
+#include <absl/container/flat_hash_map.h>
+
+namespace coproc::wasm {
+
+struct copro_deploy_info {
+    deploy_attributes attrs;
+    std::set<model::topic> topics;
+};
+
+using layout = absl::flat_hash_map<ss::sstring, copro_deploy_info>;
+
+static model::record_batch_reader
+status(model::node_id nid, layout&& lt, bool is_up) {
+    rapidjson::StringBuffer buffer;
+    rapidjson::Writer<rapidjson::StringBuffer> writer(buffer);
+    writer.StartObject();
+    writer.Key("status");
+    writer.String(is_up ? "up" : "down");
+    writer.Key("node_id");
+    writer.Int(nid());
+    writer.Key("coprocessors");
+    writer.StartObject();
```

Comment:
did you consider an array of objects (where each object had a `name` field)? It doesn't matter too much, but IMO an array of objects easier for clients to work with because iterating over dictionaries is often different in different languages.

Reply:
I made this a dictionary because each name field should be unique. Opinions?

Reply:
up to you it's a nit.